tests/test_models/test_losses/test_ce_loss.py

```diff
@@ -86,4 +198,136 @@ def test_ce_loss():
         loss_name='loss_ce')
     loss_cls = build_loss(loss_cls_cfg)
     assert loss_cls.loss_name == 'loss_ce'
+
+    # test ce loss with ignore index
+    loss_cls_cfg = dict(
+        type='CrossEntropyLoss',
+        use_sigmoid=False,
+        reduction='mean',
+        class_weight=None,
+        loss_weight=1.0,
+        avg_non_ignore=True)
+    loss_cls = build_loss(loss_cls_cfg)
+
+    fake_pred = torch.randn(2, 5, 10).float()  # 5-way classification
+    fake_label = torch.randint(0, 5, (2, 10)).long()
+    loss = loss_cls(fake_pred, fake_label)
+    torch_loss = torch.nn.functional.cross_entropy(
+        fake_pred, fake_label, reduction='mean')
+    assert torch.allclose(loss, torch_loss)
+
+    fake_label[0, [1, 2, 5, 7]] = 10  # set ignore_index
+    fake_label[1, [0, 5, 8, 9]] = 10
+    loss = loss_cls(fake_pred, fake_label, ignore_index=10)
+    torch_loss = torch.nn.functional.cross_entropy(
+        fake_pred, fake_label, ignore_index=10, reduction='mean')
+    assert torch.allclose(loss, torch_loss)
+
+    # test ignore index and class_weight
+    class_weight = torch.rand(5)
+    class_weight /= class_weight.sum()
+    loss_cls_cfg = dict(
+        type='CrossEntropyLoss',
+        use_sigmoid=False,
+        reduction='mean',
+        class_weight=class_weight,
+        loss_weight=1.0,
+        avg_non_ignore=True)
+    loss_cls = build_loss(loss_cls_cfg)
+    loss = loss_cls(fake_pred, fake_label, ignore_index=10)
+    torch_loss = torch.nn.functional.cross_entropy(
+        fake_pred,
+        fake_label,
+        ignore_index=10,
+        reduction='sum',
+        weight=class_weight) / 12.0
+    assert torch.allclose(loss, torch_loss)
+
```

Comment:
test ce loss without ignore index and `avg_non_ignore` doesn't affect the results